### ISSUE TITLE
Cannot create file observables with UTF-8 filenames

### ISSUE THREAD

**Issue by shumbert** [TheHive-Project/TheHive#780]
### Request Type
Bug

### Work Environment
| Question              | Answer
|---------------------------|--------------------
| OS version (server)       | Debian 9
| OS version (client)       | Ubuntu 16.04
| TheHive version / git hash   | 3.0.10-1
| Package Type              | Docker
| Browser type & version    | thehive4py 1.5.0

### Problem Description
When adding file observables to a case with TheHive4py, POST request gets rejected with 413 or 400 if the filename contain UTF-8 characters.

### Steps to Reproduce
Created four files: hello.txt and 你好.txt (50KB each), hello2.txt and 你好2.txt (150KB each), then used something like this to add them as file observables:
```
observable = CaseObservable(
    dataType = 'file',
    data     = ['/tmp/hello.txt'],
    tlp      = 3,
    ioc      = False,
    tags     = ['submission'],
    message  = 'Found as email attachment'
)
response = api.create_case_observable('xxxxxxxxxxxxxxxxxxxx', observable)
if response.status_code == 201:
    print('All good')
else:
    print('[ERROR] Cannot create file observable: %s (%s)' % (response.status_code, response.text))
```

Both hello.txt and hello2.txt are successfully added. 你好.txt is rejected with 400 Bad Request:
```
{"type":"BadRequest","message":"Artifact must contain data or attachment (but not both)"}
```

你好2.txt is rejected with 413 Request Entity Too Large.

### Complementary information
The only difference I noticed is in the POST request multipart/form-data. When uploading hello.txt here is the Content-Disposition header:
```
Content-Disposition: form-data; name="attachment"; filename="hello.txt"
```

And for 你好.txt:
```
Content-Disposition: form-data; name="attachment"; filename*=utf-8''%E4%BD%A0%E5%A5%BD.txt
```

My understanding both of them are valid according to RFC5987. Maybe the Play framework or the akka web server do not properly handle the latter?

**Comment by nadouani** [comment]
What happens when you add these observables from TheHive's UI?

**Comment by shumbert** [comment]
Adding 你好.txt via the UI worked fine.

**Comment by nadouani** [comment]
So this is a TheHive4py issue. I'll take a look

**Comment by nadouani** [comment]
> My understanding both of them are valid according to RFC5987. Maybe the Play framework or the akka web server do not properly handle the latter?

Yes, after some research (thanks to @3c7) it seems like the `filename*=utf-8''%E4%BD%A0%E5%A5%BD.txt` is a valid notation.

We will try to see if this can be handled by Play.

**Comment by To-om** [comment]
This issue is related to playframework/playframework#6275

**Comment by nicpenning** [comment]
This still appears to be an issue when using the alerts API and having a special character in the Title field (note that the description field handles this no problem - it makes the character a "?" symbol):

Here is a sample API call:
    $JSON = @"
    {
        "title": "Broken Title ®",
        "description": "alert Description",
        "type": "alertType",
        "source": "alertSource",
        "sourceRef": "alertSourceRef",
        "tags": "alertTag"
    }
"@
    Invoke-RestMethod -Method POST -Headers $headers -Uri $theHiveApiURLAlert -Body $JSON -ContentType "application/json"

Here is the error:
**Invoke-RestMethod : A client error occurred on POST /api/alert : Invalid Json: Invalid UTF-8 start byte 0xae**
 at [Source: akka.util.ByteIterator$ByteArrayIterator$$anon$1@1d00b160; line: 2, column: 20]
At .\alert.ps1
